### PR TITLE
Issue3/improve code

### PR DESCRIPTION
**Pull request by admy7**
A couple of improvements for the RetrieveApplicationService: 

- Improve readability 
- Nullability checks
- Add unit test for ApplicationMapper

### CODE REVIEW

**Review by inderps — DISMISSED** [review]

**Review by inderps — DISMISSED** [review]

File: src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java [hunk 1]
```diff
@@ -6,157 +6,241 @@
...
+
+        return new RetrievedApplicationWorkflow(workflow.map(Response10953Workflow::getWorkflowId)
+                .orElse(null),
+                workflow.map(Response10953Workflow::getWorkflowType).orElse(null));
     }
 
-    private static RetrievedApplicationApplicant toRetrievedApplicationApplicant(
-            UserWithAttributes remsApplicant) {
+    private RetrievedApplicationApplicant toApplicant(Application application) {
+        Optional<UserWithAttributes> applicant = Optional.ofNullable(application
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java:55]: suggestion: to improve readability, you can simply assign to `var` in all the cases.

File: src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java [hunk 2]
```diff
@@ -6,157 +6,241 @@
...
+        Optional<Set<UserWithAttributes>> potentialMembers = Optional.ofNullable(application
+                .getApplicationMembers());
+
+        return potentialMembers.map(members -> members
                 .stream()
-                .map(member -> new RetrievedApplicationMember(member.getUserid(),
-                        member.getName(),
-                        member.getEmail()))
-                .toList();
+                .map(member -> {
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java:70]: important: to improve readability, you should assign this lambda function into a anther private method, the same for the other cases.

File: src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java [hunk 3]
```diff
@@ -6,157 +6,241 @@
...
+import jakarta.enterprise.context.ApplicationScoped;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 
-public class RemsApplicationMapper {
+@ApplicationScoped
+public class RemsApplicationMapper implements ApplicationMapper<Application> {
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java:17]: important: this `RemsApplicationMapper` is a concrete class highly coupled with REMS api implementation. So it should much closer to it.

I liked the idea, but perhaps this interface should be about integration with REMS directly and not just about parsing POJOs / DTOs. I've got an idea how to fix it with clean architecture, but I would like to validate it with you.

File: src/main/java/io/github/genomicdatainfrastructure/daam/utils/ApplicationValidator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import io.github.genomicdatainfrastructure.daam.exceptions.ApplicationNotFoundException;
+import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationsApi;
+import io.github.genomicdatainfrastructure.daam.remote.rems.model.Application;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.WebApplicationException;
+
+@ApplicationScoped
+public class ApplicationValidator {
+
+    public Application validate(Long applicationId, String remsApiKey, String userId,
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/utils/ApplicationValidator.java:16]: important: this class is doing much more than validating, it's an strategy/adapter to fetch Applications from REMS. Perhaps this one should have a generic interface, and the mapper should be part of it, as composition.

Let's validate this idea together.

File: src/test/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+public class RemsApplicationMapperTest {
+
+    private static final ApplicationMapper mapper = new RemsApplicationMapper();
+
+    @Test
+    public void should_map_correctly_complete_rems_application_to_retrieved_application() {
+        RetrievedApplication retrievedApplication = mapper.from(createApplication());
+
+        assertEquals(1L, retrievedApplication.getId());
```
Comment (brunopacheco1) [inline src/test/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapperTest.java:25]: suggestion: Perhaps you could use `AssertJ` to improve the readability of this test.

**Review by brunopacheco1 — CHANGES_REQUESTED** [review]

**Review by brunopacheco1 — APPROVED** [review]

**Pull request merged**